protocols/gossipsub/src/error.rs

```diff
@@ -40,6 +40,14 @@ pub enum PublishError {
     TransformFailed(std::io::Error),
 }
 
+impl std::fmt::Display for PublishError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl std::error::Error for PublishError {}
```

Comment:
`PublishError` has two variants that contain a source, can we override `source` for these please?